core/templates/components/ck-editor-helpers/ck-editor-4-widgets.initializer.ts

```diff
@@ -107,11 +108,13 @@ export class CkEditorInitializerService {
                 var that = this;
                 var customizationArgs = {};
                 customizationArgSpecs.forEach(function(spec) {
-                  customizationArgs[spec.name] = that.data[spec.name] ||
-                    spec.default_value;
+                  customizationArgs[spec.name] = that.data[spec.name];
                 });
 
+                const defaultRTEComponent: boolean = !that.isReady();
```

Comment:
I don't quite understand; before defaultRTEComponent was set by having default values, which in some sense is just components which are newly created and not "edited" from the RTE. Here, defaultRTEComponent is set by checking if a widget is already existing on the RTE.

Reply:
Oh, I see, does "default RTE component" mean a component whose values are all set to the default value in the schema? (If so maybe we should rename it to rteComponentWithDefaultValues.)

But if so then why is it a boolean here?

Reply:
No like before, defaultRTEComponent was defined as having default values which was kinda weird as it was used in places it didn't make sense. Default RTE Component in this PR is a way to generalize widgets who haven't been created, so whether the default value is highlighted or the default value isn't will both be "default". This default RTE component is used to detect when to revert changes when canceled or show the delete button.

Reply:
I believe defaultRTEComponent should be isDefaultRTEComponent though since defaultRTEComponent is confusing, tried to adopt what was there before.

Reply:
Hm ok. Don't worry too much about what was there before -- that was from quite a few years ago when our naming might not have been as stringent. Let's try and name it properly going forward.

What is the exact definition of this thing you're calling a "default rte component" and what would be a good name for it, such that any reader of the code can easily understand conceptually what is going on?

Reply:
Then I would name `defaultRTEComponent` to `isDefaultRTEComponent`

Reply:
What is the exact definition of "default RTE component", though? (Not what it's used for, but what is it?)

Specifically: is it a component whose values are all set to the default value in the schema, or does that mean something different? (You say above that it's set by "checking if a widget is already existing on the RTE" but that is not the same thing. So the problem I have now is that I don't know what we're talking about, and we need a concrete definition here that we can base further analysis on.)

Reply:
No, "default RTE component" is just a component which is "default", meaning it wasn't edited but was newly created and has default values which can either come from the highlight or default value. If isDefaultRTEComponent doesn't make that much sense, I can call it isRTEJustCreated, but isDefaultRTEComponent makes sense to me.

Reply:
that.isReady just checks if the widget hasn't been created, which essentially means it is "newly created" or made just by clicking the toolbar button and not edited.

Reply:
OK, got it, thanks. A clarification question: Suppose a component is created and saved without edits. It's then opened again. Is it still a "default RTE component"?

Reply:
No, since in this case the user decided to save the component and it is in the RTE, meaning when being edited again there should be a delete button and cancel should not discard the component.

Reply:
OK, then let's call it isNewlyCreatedComponent. How does that sound?

Reply:
Done